### PR TITLE
Restructure paymentsTab with grid property

### LINKED ISSUE #10146: Fix BEM naming on enabledContent.js
## Test plan

https://github.com/brave/browser-laptop/pull/11525#issue-265487221

----

**Describe the issue you encountered:** update enabledContent.js to our modified BEM style (ref: https://github.com/brave/browser-laptop/blob/master/docs/style.md#defining-our-blocks-elements-and-modifiers).

- Brave Version (revision SHA): master 

- Extra QA steps:
    1. Open about:preferences#payments
    2. Enable Payments
    3. Make sure the page is loaded without breaking styles

- Any related issues:

### LINKED ISSUE #10236: Restructure paymentsTab with grid property
## Test plan

https://github.com/brave/browser-laptop/pull/10238#issue-247151904

----

**Describe the issue you encountered:** Restructure paymentsTab with display:grid. It is more simple than using display:table.

- Platform (Win7, 8, 10? macOS? Linux distro?): All

- Brave Version (revision SHA): master 

- Extra QA steps:
    1. Open `about:preferences#payments`
    2. Enable payments
    3. Make sure the elements are aligned equally on both rows and columns
    4. Disable payments
    5. Make sure the switch is placed on the same place

- Screenshot if needed:

    From:
    screenshot 2017-08-02 1 53 04

    To:
    screenshot 2017-08-02 13 34 17

    Please see the width of the pulldown and the textarea becomes equal.

- Any related issues: #10146

### LINKED ISSUE #10245: Add Panel Dropdown to styles.js
## Test plan
https://github.com/brave/browser-laptop/pull/10238#issue-247151904
1. Open about:styles#dropdowns
2. Make sure PanelDropdown appears there

----

**Describe the issue you encountered:** Add Panel Dropdown to styles.js. 

- Platform (Win7, 8, 10? macOS? Linux distro?): all

- Brave Version (revision SHA): https://github.com/brave/browser-laptop/pull/10238/commits/212f7cba5a0bd2744180f49739485a9359aa6a21

- Extra QA steps:
    1.
    2.
    3.

- Any related issues: #10236

### PR DESCRIPTION
**Pull request by luixxiul**
Closes #10236

Replace display:table with display:grid to align elements

Also:
- Update enabledContent.js to BEM style (Closes #10146)
- Add PanelDropdown
- Remove paymentStyles from payment for now (until conversion is done)
- Add Panel Dropdown to styles.js (Closes #10245)

Test Plan 1:
1. Open `about:preferences#payments`
2. Enable payments
3. Make sure the elements are aligned equally on both rows and columns
4. Disable payments
5. Make sure the switch is placed on the same place

Test Plan 2:
1. Open about:styles#dropdowns
2. Make sure PanelDropdown appears there

If this does not look changed, nothing is broken :-)

screenshot 2017-08-02 13 34 17

screenshot 2017-08-02 18 06 35

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by luixxiul** [comment]
I added `gridStyles` in the same way as I created the compact bravery panel.

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 1]
```diff
@@ -169,65 +168,46 @@ class EnabledContent extends ImmutableComponent {
...
-                  data-l10n-id={walletStatusText.id}
-                  data-l10n-args={walletStatusText.args ? JSON.stringify(walletStatusText.args) : null}
-                />
-              </td>
-            </tr>
-          </tbody>
-        </table>
+        <div className={css(gridStyles.row1col1, styles.walletBar__title)} data-l10n-id='monthlyBudget' />
+        <div className={css(gridStyles.row1col2, styles.walletBar__title)} data-l10n-id='accountBalance' />
+        <div className={css(gridStyles.row1col3)} />
```
Comment (luixxiul) [inline app/renderer/components/preferences/payment/enabledContent.js]: This is the blank item.

File: app/renderer/components/common/dropdown.js [hunk 2]
```diff
@@ -38,6 +39,12 @@ class SettingDropdown extends ImmutableComponent {
   }
 }
 
+class PanelDropdown extends ImmutableComponent {
+  render () {
+    return <FormDropdown data-isPanel='true' {...this.props} />
```
Comment (cezaraugusto) [inline app/renderer/components/common/dropdown.js]: you're actually passing a string here, so this is actually truthy and not `true`. For such cases you'd need to pass as a boolean like `data-isPanel={true}`, but standard will complain. Hidden values are evaluated to `true` anyway and make standard happy, so `<FormDropdown data-isPanel {...this.props} />` is the right take.

File: app/renderer/components/common/dropdown.js [hunk 3]
```diff
@@ -67,13 +75,21 @@ const styles = StyleSheet.create({
...
     fontSize: globalStyles.fontSize.flyoutDialog
   },
+
   settings: {
     width: '280px'
   },
+
+  settings_panel: {
+    // ref: browserButton_panelItem
+    width: '180px'
```
Comment (cezaraugusto) [inline app/renderer/components/common/dropdown.js]: if `browserButton_panelItem` shares the same value then it should be both hosted in `global.js`, then the comment can be removed

File: app/renderer/components/preferences/payment/disabledContent.js [hunk 4]
```diff
@@ -59,7 +58,7 @@ const styles = StyleSheet.create({
     backgroundColor: globalStyles.color.lightGray,
     borderRadius: globalStyles.radius.borderRadiusUIbox,
     padding: '40px',
-    fontSize: paymentStyles.font.regular,
+    fontSize: globalStyles.payments.fontSize.regular,
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/payment/disabledContent.js]: why are we changing styles here?

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 5]
```diff
@@ -84,11 +83,10 @@ class EnabledContent extends ImmutableComponent {
 
     return <section className={css(styles.balance)}>
       <FormTextbox data-test-id='fundsAmount' readOnly value={btcToCurrencyString(value, ledgerData)} />
-      <a className={css(styles.iconLink)} href='https://brave.com/Payments_FAQ.html' target='_blank'>
+      <a className={css(styles.balance__iconLink)} href='https://brave.com/Payments_FAQ.html' target='_blank'>
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/payment/enabledContent.js]: ++ much better

File: app/renderer/components/preferences/payment/enabledContent.js [hunk 6]
```diff
@@ -237,85 +216,94 @@ class EnabledContent extends ImmutableComponent {
...
+    gridColumn: 3
+  }
+})
 
-    ':hover': {
-      textDecoration: 'none'
-    }
+const styles = StyleSheet.create({
+  walletBar: {
+    display: 'grid',
```
Comment (cezaraugusto) [inline app/renderer/components/preferences/payment/enabledContent.js]: yessssssss

File: app/renderer/components/common/dropdown.js (hunk 2 above)
Comment (luixxiul) [inline app/renderer/components/common/dropdown.js]: > <FormDropdown data-isPanel {...this.props} /> is the right take.

OK I'm changing.

File: app/renderer/components/preferences/payment/disabledContent.js (hunk 4 above)
Comment (luixxiul) [inline app/renderer/components/preferences/payment/disabledContent.js]: To remove `paymentsStyles` for now. It is not maintained well and moving the values to global.js makes it the code much more understandable.

I'm going to re-introduce `paymentsStyles` after files related with about:preferences#payments are polished and the other stuff (which could be related with them) on other files than ones under `payment` is also converted to normalize the common styles, if any.

Comment (luixxiul) [inline app/renderer/components/preferences/payment/disabledContent.js]: If `paymentStyles` would be left, there would remain only [this line](https://github.com/brave/browser-laptop/pull/10238/files#diff-6dd8c50f92b9abd7a2553f4f3a8b1433L9) as other two will be removed. It does not make much sense to have a const for one style, IMO.

File: app/renderer/components/styles/global.js [hunk 7]
```diff
@@ -327,6 +328,13 @@ const globalStyles = {
         background: '#ccc'
       }
     }
+  },
+
+  // TODO (Suguru): move them to payment.js after style refactoring is done
```
Comment (luixxiul) [inline app/renderer/components/styles/global.js]: @cezaraugusto this TODO will be revisited.

File: app/renderer/components/preferences/payment/disabledContent.js (hunk 4 above)
Comment (cezaraugusto) [inline app/renderer/components/preferences/payment/disabledContent.js]: ok I'm against `payments.js` styles anyway so doing what you did is better up to a point that we remove that file.

File: app/renderer/components/styles/global.js (hunk 7 above)
Comment (cezaraugusto) [inline app/renderer/components/styles/global.js]: ok thanks. also pls move things out of that most you can. I think that file should be removed

Comment (luixxiul) [inline app/renderer/components/styles/global.js]: I see. I'm going to address that on another PR (as it is out of scope of this PR and it'll take some time).

File: app/renderer/components/common/dropdown.js (hunk 2 above)
Comment (luixxiul) [inline app/renderer/components/common/dropdown.js]: Fixed with https://github.com/brave/browser-laptop/pull/10238/commits/91de03c042d85552b3a072d2569c57a01b1eaa27#diff-b906e73e687bce1b6e897d2b8d80089bR44

File: app/renderer/components/common/dropdown.js (hunk 3 above)
Comment (luixxiul) [inline app/renderer/components/common/dropdown.js]: Fixed with:
https://github.com/brave/browser-laptop/pull/10238/commits/91de03c042d85552b3a072d2569c57a01b1eaa27#diff-b906e73e687bce1b6e897d2b8d80089bR89
https://github.com/brave/browser-laptop/pull/10238/commits/91de03c042d85552b3a072d2569c57a01b1eaa27#diff-f14732b3ea914a26c2215676e7208907R310

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 5 above)
Comment (luixxiul) [inline app/renderer/components/preferences/payment/enabledContent.js]: Added `rel='noopener'` as well:
https://github.com/brave/browser-laptop/pull/10238/commits/91de03c042d85552b3a072d2569c57a01b1eaa27#diff-5d8199641ea1ba12e0394859f7eb0c55R86

File: app/renderer/components/preferences/payment/enabledContent.js (hunk 6 above)
Comment (luixxiul) [inline app/renderer/components/preferences/payment/enabledContent.js]: Grid rules 🦁

**Comment by cezaraugusto** [comment]
I tested before the conflict, can you rebase please? we can merge after that

**Comment by luixxiul** [comment]
@cezaraugusto rebased. this is ready for re-review, thanks.

**Comment by NejcZdovc** [comment]
@luixxiul can you rebase again pls

**Comment by srirambv** [comment]
@luixxiul is this expected ?
image

**Comment by luixxiul** [comment]
the right column should be aligned to the gray box. something must have been fallen out during 0.19 series. will check out.

**Pull request merged**